bugbot/crash/analyzer.py

```diff
@@ -0,0 +1,710 @@
+        elif op_sys.startswith("Windows"):
+            op_sys = "Windows"
+        elif "Linux" in op_sys or op_sys.startswith("Ubuntu"):
+            op_sys = "Linux"
+        else:
+            op_sys = "Other"
+
+        return op_sys
+
+    @property
+    def bugzilla_op_sys(self) -> str:
+        """The name of the OS where the crash happens.
+
+        The value is one of the legal values for Bugzilla's `op_sys` field.
+
+        - If no OS name is found, the value will be "Unspecified".
+        - If the OS name is not recognized, the value will be "Other".
+        - If multiple OS names are found, the value will "All". Unless the OS
+          names can be resolved common name without a version. For example,
+          "Windows 10" and "Windows 7" will become "Windows".
+        """
+        all_op_sys = {
+            self.to_bugzilla_op_sys(op_sys["term"])
+            for op_sys in self.signature["facets"]["platform_pretty_version"]
+        }
+
+        if len(all_op_sys) > 1:
+            # Resolve to root OS name by removing the version number.
+            all_op_sys = {op_sys.split(" ")[0] for op_sys in all_op_sys}
+
+        if len(all_op_sys) == 2 and "Other" in all_op_sys:
+            # TODO: explain this workaround.
+            all_op_sys.remove("Other")
+
+        if len(all_op_sys) == 1:
+            return next(iter(all_op_sys))
+
+        if len(all_op_sys) == 0:
+            return "Unspecified"
+
+        return "All"
+
+    @classmethod
+    def to_bugzilla_cpu(cls, cpu: str) -> str:
+        """Return the corresponding CPU name in Bugzilla for the provided name
+        from Socorro.
+
+        If the CPU is not recognized, return "Other".
+        """
+        if cls._bugzilla_cpu_legal_values_map is None:
+            cls._bugzilla_cpu_legal_values_map = {
+                value.lower(): value
+                for value in bugzilla.BugFields.fetch_field_values("rep_platform")
+            }
+
+        return cls._bugzilla_cpu_legal_values_map.get(cpu, "Other")
+
+    @property
+    def bugzilla_cpu_arch(self) -> str:
+        """The CPU architecture of the devices where the crash happens.
+
+        The value is one of the legal values for Bugzilla's `rep_platform` field.
+
+        - If no CPU architecture is found, the value will be "Unspecified".
+        - If the CPU architecture is not recognized, the value will be "Other".
+        - If multiple CPU architectures are found, the value will "All".
+        """
+        all_cpu_arch = {
+            self.to_bugzilla_cpu(cpu["term"])
+            for cpu in self.signature["facets"]["cpu_arch"]
+        }
+
+        if len(all_cpu_arch) == 2 and "Other" in all_cpu_arch:
+            all_cpu_arch.remove("Other")
+
+        if len(all_cpu_arch) == 1:
+            return next(iter(all_cpu_arch))
+
+        if len(all_cpu_arch) == 0:
+            return "Unspecified"
+
+        return "All"
+
+    @property
+    def num_user_comments(self) -> int:
```

Comment:
We had an experimental tool to get the most frequent words from comments, might be interesting to consider using it in the future (https://github.com/marco-c/crashcorrelations/blob/master/comments.py)

Reply:
Nice idea! I filed a follow-up issue: https://github.com/mozilla/bugbot/issues/2132.